app/code/Magebit/Faq/Model/QuestionRepository.php

```diff
@@ -0,0 +1,151 @@
+
+class QuestionRepository implements QuestionRepositoryInterface
+{
+    /**
+     * @var QuestionFactory
+     */
+    protected QuestionFactory $objectFactory;
+
+    /**
+     * @var ObjectResourceModel
+     */
+    protected ObjectResourceModel $objectResourceModel;
+
+    /**
+     * @var CollectionFactory
+     */
+    protected CollectionFactory $collectionFactory;
+
+    /**
+     * @var SearchResultsInterfaceFactory
+     */
+    protected SearchResultsInterfaceFactory $searchResultsFactory;
+
+    protected CollectionProcessor $collectionProcessor;
+
+    /**
+     * @param QuestionFactory $objectFactory
+     * @param ObjectResourceModel $objectResourceModel
+     * @param CollectionFactory $collectionFactory
+     * @param SearchResultsInterfaceFactory $searchResultsFactory
+     * @param CollectionProcessor $collectionProcessor
+     */
+    public function __construct(
+        QuestionFactory $objectFactory,
+        ObjectResourceModel $objectResourceModel,
+        CollectionFactory $collectionFactory,
+        SearchResultsInterfaceFactory $searchResultsFactory,
+        CollectionProcessor $collectionProcessor
+    ) {
+        $this->objectFactory = $objectFactory;
+        $this->objectResourceModel = $objectResourceModel;
+        $this->collectionFactory = $collectionFactory;
+        $this->searchResultsFactory = $searchResultsFactory;
+        $this->collectionProcessor = $collectionProcessor;
+    }
+
+    /**
+     * Get question by ID.
+     *
+     * @param int $id
+     * @return QuestionInterface
+     * @throws NoSuchEntityException
+     */
+    public function get( $id): QuestionInterface
+    {
+        $object = $this->objectFactory->create();
+        $this->objectResourceModel->load($object, $id);
+        if (!$object->getId()) {
+            throw new NoSuchEntityException(__('Object with id "%1" does not exist.', $id));
+        }
+        return $object;
+    }
+
+    /**
+     * Save a question.
+     *
+     * @param QuestionInterface $question
+     * @return QuestionInterface
+     * @throws CouldNotSaveException
+     */
+    public function save(QuestionInterface $question): QuestionInterface
+    {
+        try {
+            $this->objectResourceModel->save($question);
+
+        } catch (\Exception $e)
+        {
+            throw new CouldNotSaveException(__('Cannot save the question'));
+        }
+        return $question;
+    }
+
+    /**
+     * Retrieve a list of questions based on search criteria.
+     *
+     * @param SearchCriteriaInterface $searchCriteria
+     * @return SearchResultsInterface
+     */
+    public function getList(SearchCriteriaInterface $searchCriteria): SearchResultsInterface
+    {
+        $collection = $this->collectionFactory->create();
+        $this->collectionProcessor->process($searchCriteria, $collection);
+        $searchResults = $this->searchResultsFactory->create();
+        $searchResults->setSearchCriteria($searchCriteria);
```

Comment:
This is just a preference, but you can also do 
$searchResults->setSearchCriteria($searchCriteria)
                         ->setItems($collection->getItems())
                         ->setTotalCount($collection->getItems());